crates/ruff_linter/src/rules/flake8_pytest_style/rules/parametrize.rs

```diff
@@ -81,13 +82,41 @@ impl Violation for PytestParametrizeNamesWrongType {
 
     #[derive_message_formats]
     fn message(&self) -> String {
-        let PytestParametrizeNamesWrongType { expected } = self;
-        format!("Wrong name(s) type in `@pytest.mark.parametrize`, expected `{expected}`")
+        let PytestParametrizeNamesWrongType {
+            single_argument,
+            expected,
+        } = self;
+        let expected_string;
+        if *single_argument {
+            expected_string = "`str`".to_string();
+        } else {
+            match expected {
+                types::ParametrizeNameType::Csv => expected_string = format!("a {expected}"),
+                types::ParametrizeNameType::Tuple | types::ParametrizeNameType::List => {
+                    expected_string = format!("`{expected}`");
+                }
+            }
+        }
```

Comment:
I'd write this as so:

```suggestion
        let expected_string = {
	        if *single_argument {
	            "`str`".to_string();
	        } else {
	            match expected {
	                types::ParametrizeNameType::Csv => format!("a {expected}"),
	                types::ParametrizeNameType::Tuple | types::ParametrizeNameType::List => {
	                    format!("`{expected}`");
	                }
	            }
	        }
	    };
```